pilot/helpers/agents/CodeMonkey.py

```diff
@@ -296,118 +170,180 @@ def identify_file_to_change(self, code_changes_description: str, files: list[dic
         }, GET_FILE_TO_MODIFY)
         return llm_response["file"]
 
-    @staticmethod
-    def get_code_blocks(llm_response: str) -> list[tuple[str, str]]:
+    def review_change(
+        self,
+        convo: AgentConvo,
+        instructions: str,
+        file_name: str,
+        old_content: str,
+        new_content: str
+    ) -> str:
         """
-        Split the response into code block(s).
+        Review changes that were applied to the file.
 
-        Ignores any content outside of code blocks.
+        This asks the LLM to act as a PR reviewer and for each part (hunk) of the
+        diff, decide if it shold be applied (kept) or ignored (removed from the PR).
 
-        :param llm_response: response from the LLM
-        :return: list of pairs of current and new blocks
+        :param convo: AgentConvo instance
+        :param instructions: instructions for the reviewer
+        :param file_name: name of the file being modified
+        :param old_content: old file content
+        :param new_content: new file content (with proposed changes)
+        :return: file content update with approved changes
+
+        Diff hunk explanation: https://www.gnu.org/software/diffutils/manual/html_node/Hunks.html
         """
-        pattern = re.compile(
-            r"CURRENT_CODE:\n```([a-z0-9]+)?\n(.*?)\n```\nNEW_CODE:\n```([a-z0-9]+)?\n(.*?)\n?```\nEND\s*",
-            re.DOTALL
-        )
-        pairs = []
-        for block in pattern.findall(llm_response):
-            pairs.append((block[1], block[3]))
-        return pairs
+
+        hunks = self.get_diff_hunks(file_name, old_content, new_content)
+
+        llm_response = convo.send_message('development/review_changes.prompt', {
+            "instructions": instructions,
+            "file_name": file_name,
+            "old_content": old_content,
+            "hunks": hunks,
+        }, REVIEW_CHANGES)
+
+        hunk_ids = set()
+        for hunk in llm_response.get("hunks", []):
```

Comment:
before doing this for loop i would add a check if `len(hunks) == len(llm_response.get("hunks", []))` and if not i would retry send_message()